thunder/executors/transformer_engineex.py

```diff
@@ -333,7 +332,10 @@ def _te_functional_linear_meta(
 
             # It's not critical to model the exact shape and dtype of
             # saved_tensors since they are not used in Thunder's meta functions.
-            saved_tensors = tuple(TensorProxy(like=a, shape=a.shape) for _ in range(_get_num_saved_tensors()))
+            saved_tensors = tuple(
+                TensorProxy(like=a, shape=a.shape)
+                for _ in range(_get_num_saved_tensors(get_recipe_from_options_or_default_recipe()))
```

Comment:
this is more of a design discussion (so, not blocking this PR), I'm wondering if we should make executor-specific options more discoverable for developers by adding them as arguments to the executor itself

In general, transforms do expose arguments, and executors can (and should imo) do it as well.

We can keep the compile options mechanism just like it is today, and have compile options override transform and executor arguments if specified.

Reply:
Certainly, sounds good @IvanYashchuk